reactor-netty-core/src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -138,7 +141,12 @@ protected PooledConnectionProvider(Builder builder) {
 			List<Mono<Void>> pools;
 			pools = channelPools.values()
 			                    .stream()
-			                    .map(Pool::disposeLater)
+			                    .map(pool -> {
+			                        if (pool instanceof GracefulShutdownInstrumentedPool) {
+			                            return ((GracefulShutdownInstrumentedPool<T>) pool).disposeGracefully(disposeTimeout);
```

Comment:
maybe we want to `onErrorResume` this, since unlike `pool.disposeLater()` the graceful shutdown can "fail" (it just notifies that the resources were not all `released()` within the provided `Duration`, but does so after having actually shut down the pool anyway).